compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -164,6 +165,50 @@ void update_quantparam(record_minmax::MinMaxObserver *observer, const std::strin
   }
 }
 
+std::unordered_set<std::string> list_regular_files(DIR *directory)
```

Comment:
any particular reason to must use `unordered_set` ?

Reply:
As the function will be used uniquely in this file it seems reasonable to return a set, because we need a set anyway to implement readable and fast search of file names. std::unordered_set seemingly fits the task best, but the performance boost compared to std::set is insufficient (if there is any), so I can use std::set instead for the sake of readability.